File: sdkwork-ai-sdk-swift/Sources/API/ThreadsApi.swift
import Foundation

public class ThreadsApi {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// Submit tool outputs
    public func submitToolOutputs(thread_id: String, run_id: String, body: SubmitToolOutputsRequest) async throws -> SubmitToolOutputsResponse? {
        let response = try await client.post(ApiPaths.aiPath("/threads/\(thread_id)/runs/\(run_id)/submit_tool_outputs"), body: body)
        return response as? SubmitToolOutputsResponse
    }

    /// Cancel run
    public func cancelRun(thread_id: String, run_id: String) async throws -> RunResponse? {
        let response = try await client.post(ApiPaths.aiPath("/threads/\(thread_id)/runs/\(run_id)/cancel"), body: nil)
        return response as? RunResponse
    }

    /// Get run
    public func getRun(thread_id: String, run_id: String) async throws -> RunResponse? {
        let response = try await client.get(ApiPaths.aiPath("/threads/\(thread_id)/runs/\(run_id)"))
        return response as? RunResponse
    }

    /// Modify run
    public func updateRun(thread_id: String, run_id: String, body: UpdateRunRequest) async throws -> RunResponse? {
        let response = try await client.post(ApiPaths.aiPath("/threads/\(thread_id)/runs/\(run_id)"), body: body)
        return response as? RunResponse
    }

    /// List runs
    public func listRuns(thread_id: String, params: [String: Any]? = nil) async throws -> RunListResponse? {
        let response = try await client.get(ApiPaths.aiPath("/threads/\(thread_id)/runs"), params: params)
        return response as? RunListResponse
    }

    /// Create run
    public func createRun(thread_id: String, body: RunCreateRequest) async throws -> CreateRunResponse? {
        let response = try await client.post(ApiPaths.aiPath("/threads/\(thread_id)/runs"), body: body)
        return response as? CreateRunResponse
    }

    /// Get thread message
    public func getThreadMessage(thread_id: String, message_id: String) async throws -> ThreadMessageResponse? {
        let response = try await client.get(ApiPaths.aiPath("/threads/\(thread_id)/messages/\(message_id)"))
        return response as? ThreadMessageResponse
    }

    /// Modify thread message
    public func updateThreadMessage(thread_id: String, message_id: String, body: ThreadMessageModifyRequest) async throws -> ThreadMessageResponse? {
        let response = try await client.post(ApiPaths.aiPath("/threads/\(thread_id)/messages/\(message_id)"), body: body)
        return response as? ThreadMessageResponse
    }

    /// List thread messages
    public func listThreadMessages(thread_id: String, params: [String: Any]? = nil) async throws -> ThreadMessageListResponse? {
        let response = try await client.get(ApiPaths.aiPath("/threads/\(thread_id)/messages"), params: params)
        return response as? ThreadMessageListResponse
    }

    /// Create thread message
    public func createThreadMessage(thread_id: String, body: ThreadMessageCreateRequest) async throws -> ThreadMessageResponse? {
        let response = try await client.post(ApiPaths.aiPath("/threads/\(thread_id)/messages"), body: body)
        return response as? ThreadMessageResponse
    }

    /// Get thread
    public func getThread(thread_id: String) async throws -> ThreadResponse? {
        let response = try await client.get(ApiPaths.aiPath("/threads/\(thread_id)"))
        return response as? ThreadResponse
    }

    /// Update thread
    public func updateThread(thread_id: String, body: ThreadCreateRequest) async throws -> ThreadResponse? {
        let response = try await client.post(ApiPaths.aiPath("/threads/\(thread_id)"), body: body)
        return response as? ThreadResponse
    }

    /// Delete thread
    public func deleteThread(thread_id: String) async throws -> ThreadDeleteResponse? {
        let response = try await client.delete(ApiPaths.aiPath("/threads/\(thread_id)"))
        return response as? ThreadDeleteResponse
    }

    /// Create thread and run
    public func createThreadAndRun(body: ThreadRunCreateRequest) async throws -> CreateThreadAndRunResponse? {
        let response = try await client.post(ApiPaths.aiPath("/threads/runs"), body: body)
        return response as? CreateThreadAndRunResponse
    }

    /// Create thread
    public func createThread(body: ThreadCreateRequest) async throws -> ThreadResponse? {
        let response = try await client.post(ApiPaths.aiPath("/threads"), body: body)
        return response as? ThreadResponse
    }

    /// Get run step
    public func getRunStep(thread_id: String, run_id: String, step_id: String) async throws -> RunStepResponse? {
        let response = try await client.get(ApiPaths.aiPath("/threads/\(thread_id)/runs/\(run_id)/steps/\(step_id)"))
        return response as? RunStepResponse
    }

    /// List run steps
    public func listRunSteps(thread_id: String, run_id: String, params: [String: Any]? = nil) async throws -> RunStepListResponse? {
        let response = try await client.get(ApiPaths.aiPath("/threads/\(thread_id)/runs/\(run_id)/steps"), params: params)
        return response as? RunStepListResponse
    }
}

File: sdkwork-ai-sdk-csharp/Api/ThreadsApi.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Ai.Http;
using Ai.Models;

namespace Ai.Api
{
    public class ThreadsApi
    {
        private readonly HttpClient _client;

        public ThreadsApi(HttpClient client)
        {
            _client = client;
        }

        /// <summary>
        /// Submit tool outputs
        /// </summary>
        public async Task<SubmitToolOutputsResponse?> SubmitToolOutputsAsync(string thread_id, string run_id, SubmitToolOutputsRequest body)
        {
            return await _client.PostAsync<SubmitToolOutputsResponse>(ApiPaths.AiPath($"/v1/threads/{thread_id}/runs/{run_id}/submit_tool_outputs"), body);
        }

        /// <summary>
        /// Cancel run
        /// </summary>
        public async Task<RunResponse?> CancelRunAsync(string thread_id, string run_id)
        {
            return await _client.PostAsync<RunResponse>(ApiPaths.AiPath($"/v1/threads/{thread_id}/runs/{run_id}/cancel"), null);
        }

        /// <summary>
        /// Get run
        /// </summary>
        public async Task<RunResponse?> GetRunAsync(string thread_id, string run_id)
        {
            return await _client.GetAsync<RunResponse>(ApiPaths.AiPath($"/v1/threads/{thread_id}/runs/{run_id}"));
        }

        /// <summary>
        /// Modify run
        /// </summary>
        public async Task<RunResponse?> UpdateRunAsync(string thread_id, string run_id, UpdateRunPostRequest body)
        {
            return await _client.PostAsync<RunResponse>(ApiPaths.AiPath($"/v1/threads/{thread_id}/runs/{run_id}"), body);
        }

        /// <summary>
        /// List runs
        /// </summary>
        public async Task<RunListResponse?> ListRunsAsync(string thread_id, Dictionary<string, object>? query = null)
        {
            return await _client.GetAsync<RunListResponse>(ApiPaths.AiPath($"/v1/threads/{thread_id}/runs"), query);
        }

        /// <summary>
        /// Create run
        /// </summary>
        public async Task<CreateRunResponse?> CreateRunAsync(string thread_id, RunCreateRequest body)
        {
            return await _client.PostAsync<CreateRunResponse>(ApiPaths.AiPath($"/v1/threads/{thread_id}/runs"), body);
        }

        /// <summary>
        /// Get thread message
        /// </summary>
        public async Task<ThreadMessageResponse?> GetThreadMessageAsync(string thread_id, string message_id)
        {
            return await _client.GetAsync<ThreadMessageResponse>(ApiPaths.AiPath($"/v1/threads/{thread_id}/messages/{message_id}"));
        }

        /// <summary>
        /// Modify thread message
        /// </summary>
        public async Task<ThreadMessageResponse?> UpdateThreadMessageAsync(string thread_id, string message_id, ThreadMessageModifyRequest body)
        {
            return await _client.PostAsync<ThreadMessageResponse>(ApiPaths.AiPath($"/v1/threads/{thread_id}/messages/{message_id}"), body);
        }

        /// <summary>
        /// List thread messages
        /// </summary>
        public async Task<ThreadMessageListResponse?> ListThreadMessagesAsync(string thread_id, Dictionary<string, object>? query = null)
        {
            return await _client.GetAsync<ThreadMessageListResponse>(ApiPaths.AiPath($"/v1/threads/{thread_id}/messages"), query);
        }

        /// <summary>
        /// Create thread message
        /// </summary>
        public async Task<ThreadMessageResponse?> CreateThreadMessageAsync(string thread_id, ThreadMessageCreateRequest body)
        {
            return await _client.PostAsync<ThreadMessageResponse>(ApiPaths.AiPath($"/v1/threads/{thread_id}/messages"), body);
        }

        /// <summary>
        /// Get thread
        /// </summary>
        public async Task<ThreadResponse?> GetThreadAsync(string thread_id)
        {
            return await _client.GetAsync<ThreadResponse>(ApiPaths.AiPath($"/v1/threads/{thread_id}"));
        }

        /// <summary>
        /// Update thread
        /// </summary>
        public async Task<ThreadResponse?> UpdateThreadAsync(string thread_id, ThreadCreateRequest body)
        {
            return await _client.PostAsync<ThreadResponse>(ApiPaths.AiPath($"/v1/threads/{thread_id}"), body);
        }

        /// <summary>
        /// Delete thread
        /// </summary>
        public async Task<ThreadDeleteResponse?> DeleteThreadAsync(string thread_id)
        {
            return await _client.DeleteAsync<ThreadDeleteResponse>(ApiPaths.AiPath($"/v1/threads/{thread_id}"));
        }

        /// <summary>
        /// Create thread and run
        /// </summary>
        public async Task<CreateThreadAndRunPostResponse?> CreateThreadAndRunAsync(ThreadRunCreateRequest body)
        {
            return await _client.PostAsync<CreateThreadAndRunPostResponse>(ApiPaths.AiPath("/v1/threads/runs"), body);
        }

        /// <summary>
        /// Create thread
        /// </summary>
        public async Task<ThreadResponse?> CreateThreadAsync(ThreadCreateRequest body)
        {
            return await _client.PostAsync<ThreadResponse>(ApiPaths.AiPath("/v1/threads"), body);
        }

        /// <summary>
        /// Get run step
        /// </summary>
        public async Task<RunStepResponse?> GetRunStepAsync(string thread_id, string run_id, string step_id)
        {
            return await _client.GetAsync<RunStepResponse>(ApiPaths.AiPath($"/v1/threads/{thread_id}/runs/{run_id}/steps/{step_id}"));
        }

        /// <summary>
        /// List run steps
        /// </summary>
        public async Task<RunStepListResponse?> ListRunStepsAsync(string thread_id, string run_id, Dictionary<string, object>? query = null)
        {
            return await _client.GetAsync<RunStepListResponse>(ApiPaths.AiPath($"/v1/threads/{thread_id}/runs/{run_id}/steps"), query);
        }
    }
}
